app/src/main/java/at/aau/edu/wizards/MainActivity.kt

```diff
@@ -43,15 +49,50 @@ class MainActivity : AppCompatActivity() {
     }
 
     private fun setupUi() {
+        val sharedPrefs: SharedPreferences = getSharedPreferences(getString(R.string.shared_storage), Context.MODE_PRIVATE)
+        if (usernameAlreadyExists(sharedPrefs)) {
+            binding.inputUsername.setText(sharedPrefs.getString(SHARED_PREFERENCE_USERNAME_KEY, ""))
+            setButtonsEnableStatus(true)
+        }
+        else {
+            setButtonsEnableStatus(false)
+        }
+
+        binding.inputUsername.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
+            override fun onTextChanged(typedText: CharSequence, i: Int, i1: Int, i2: Int) {
+                if (typedText.isNotEmpty()) {
+                    setButtonsEnableStatus(true)
+                }
+                else {
+                    setButtonsEnableStatus(false)
+                }
+            }
+
+            override fun afterTextChanged(editable: Editable) {}
+        })
         binding.clientButton.setOnClickListener {
-            showDiscoverFragment()
+            if (binding.clientButton.isEnabled) {
+                showDiscoverFragment()
+            }
```

Comment:
here i would also add an else that for example shows a toast "please enter a username" when the user clicks on the greyed out button.

Reply:
Code not 100% accurate, but you can also show Snackbar instead. Error case handling is important, always try to keep in mind that users can be "really stupid"

`Snackbar.makeText(context, R.string.empty_username_warning, Snackbar.LengthLong).show()`

Reply:
As the button is disabled until the username is given, the onClick event does not fire, therefore, we cannot detect a click on the disabled button. We can do this by displaying a label to inform the user that they first need to enter username to proceed.